### PR TITLE
Feature/firebase login operations

### PR DESCRIPTION
**Pull request by burakkoc5**
Firebase authentication setup for signin, signup.
A login button that triggers the signin operation.
On success, the user is redirected to the home page.
On failure, an error message is displayed.
State management has been implemented to manage login states with Bloc.
Minor UI adjustments, such as button styling changes.

### CODE REVIEW

File: android/app/google-services.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
```
Comment (E-MRE) [inline android/app/google-services.json]: gitignore ile bu dosyayı gizlemekte yarar var. dışarıdan görünmesi güvenlik açısından tehlike

File: firebase.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{"flutter":{"platforms":{"android":{"default":{"projectId":"asset-tracker-3cc21","appId":"1:490137200342:android:52806d2e5cffbb33210b77","fileOutput":"android/app/google-services.json"}},"ios":{"default":{"projectId":"asset-tracker-3cc21","appId":"1:490137200342:ios:32fa08ba5eeb1236210b77","uploadDebugSymbols":false,"fileOutput":"ios/Runner/GoogleService-Info.plist"}},"macos":{"default":{"projectId":"asset-tracker-3cc21","appId":"1:490137200342:ios:32fa08ba5eeb1236210b77","uploadDebugSymbols":false,"fileOutput":"macos/Runner/GoogleService-Info.plist"}},"dart":{"lib/firebase_options.dart":{"projectId":"asset-tracker-3cc21","configurations":{"android":"1:490137200342:android:52806d2e5cffbb33210b77","ios":"1:490137200342:ios:32fa08ba5eeb1236210b77","macos":"1:490137200342:ios:32fa08ba5eeb1236210b77","web":"1:490137200342:web:383dfe2c4b40ac13210b77","windows":"1:490137200342:web:6c77bd6fbb179f85210b77"}}}}}}
```
Comment (E-MRE) [inline firebase.json]: bunu da gizleyebiliriz.

File: lib/firebase_options.dart [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+///
+/// Example:
+/// ```dart
+/// import 'firebase_options.dart';
+/// // ...
+/// await Firebase.initializeApp(
+///   options: DefaultFirebaseOptions.currentPlatform,
+/// );
+/// ```
+class DefaultFirebaseOptions {
```
Comment (E-MRE) [inline lib/firebase_options.dart]: bu dosya da gizlenebilir

**Review by E-MRE — CHANGES_REQUESTED** [review]
Güzel olmuş, önemli dosyaları gizlersen çok daha iyi olacak

File: lib/core/widgets/container_button.dart [hunk 4]
```diff
@@ -33,26 +33,29 @@ class AppContainerButton extends StatelessWidget {
...
+      ),
+      child: Material(
+        color: Colors.transparent,
+        child: InkWell(
+          onTap: onPressed,
+          splashColor: Theme.of(context).colorScheme.primary.withOpacity(0.7),
+          borderRadius: Radiuses.sm.all,
+          child: Container(
+            padding: padding ??
+                EdgeInsets.only(left: isBig ? 16 : 8, right: isBig ? 16 : 4),
```
Comment (E-MRE) [inline lib/core/widgets/container_button.dart]: sizes

File: lib/main.dart [hunk 5]
```diff
@@ -1,13 +1,26 @@
...
+import 'package:asset_tracker/firebase_options.dart';
 import 'package:asset_tracker/i18n/strings.g.dart';
+import 'package:firebase_core/firebase_core.dart';
 import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
 
-void main() {
+void main() async {
   WidgetsFlutterBinding.ensureInitialized();
+  await Firebase.initializeApp(
```
Comment (E-MRE) [inline lib/main.dart]: appInıt

File: lib/main.dart [hunk 6]
```diff
@@ -1,13 +1,26 @@
...
+    options: DefaultFirebaseOptions.currentPlatform,
+  );
+  setupLocator();
   LocaleSettings.useDeviceLocale();
-  runApp(const MyApp());
+
+  runApp(MultiBlocProvider(providers: [
+    BlocProvider<AuthenticationCubit>(
```
Comment (E-MRE) [inline lib/main.dart]: multi provider

**Review by E-MRE — APPROVED** [review]
Eline sağlık

**Comment by burakkoc5** [comment]
### Merge Request: Authentication & Firebase Enhancements  

- **Firebase Integration:**  
  - Configured Firebase for Android/iOS.  
  - Added and later removed sensitive Firebase files from version control.  
  - Created `FirebaseAuthException` extension for error handling.  

- **Authentication Features:**  
  - Implemented sign-in, sign-up, and sign-out functionality.  
  - Integrated `AuthCubit` for state management during registration.  

- **UI Enhancements:**  
  - Fixed overflow issue on registration forms with a scrollable view.  
  - Added ripple effect for button interactions.  

- **Localization:**  
  - Added translations for all user-facing strings.  

- **Codebase Improvements:**  
  - Updated `.gitignore` to exclude sensitive files.  
  - Integrated `GetIt` for service locator pattern.  
  - Edited folder structure for better organization.  

- **Bug Fixes:**  
  - Addressed minor UI and functionality issues.  
  - Cleaned up and secured version control files.

**Pull request merged**